### PR TITLE
*: update all dependencies of golang and rust to newest version

### PR DESCRIPTION
**Pull request by elenaf9**
Update dependencies:

**golang/:**
- Use libp2p from go-libp2p monorepo, v0.23.4
- Update go-multiaddr v0.3.3 -> v0.8.0

**rust/:**
- Update libp2p v0.45.0 -> v0.50.0
- Update async-std v1.6.2 -> 1.12.0
- Update bytes v1.0.1 -> 1.3.0

### CODE REVIEW

**Comment by mxinden** [comment]
>   thread 'main' panicked at 'Could not find `protoc` installation and this build crate cannot proceed without

Can you add the `protoc` installation step to CI in this pull request @elenaf9?

File: .github/workflows/makefile.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
 
       - uses: actions/setup-go@v2
         with:
-          go-version: "1.16.x"
+          go-version: "1.19.x"
```
Comment (elenaf9) [inline .github/workflows/makefile.yml:18]: `quic-go` requires go version >1.18.0, thus the `makefile.yml` CI job fails with `go-version: "1.16.x"`. I am not familiar with workflows for golang; is it enough to just update the version here?

File: .github/workflows/makefile.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ jobs:
```
Comment (mxinden) [inline .github/workflows/makefile.yml:15]:
```suggestion
      - name: Install Protoc
        uses: arduino/setup-protoc@64c0c85d18e984422218383b81c52f8b077404d3 # v1.1.2
```

**Comment by mxinden** [comment]
Thanks @elenaf9 for the help here.

**Pull request merged**